websites/Y/YomuMangás/presence.ts

```diff
@@ -0,0 +1,162 @@
+const presence = new Presence({ clientId: "869377195682983957" }),
+	readingTimestamp = Math.floor(Date.now() / 1000);
+
+presence.on("UpdateData", async () => {
+	const { pathname } = window.location,
+		presenceData: PresenceData = { largeImageKey: "logo" },
+		[images, manga, chapter, user, timestamp, adult, privacy] =
+			await Promise.all([
+				presence.getSetting<boolean>("images"),
+				presence.getSetting<boolean>("manga"),
+				presence.getSetting<boolean>("chapter"),
+				presence.getSetting<boolean>("user"),
+				presence.getSetting<boolean>("timestamp"),
+				presence.getSetting<boolean>("adult"),
+				presence.getSetting<boolean>("privacy"),
+			]);
+
+	if (privacy) {
+		presenceData.details = "Criando um novo mundo!";
+		presenceData.state = "yomumangas.com";
+		if (timestamp) presenceData.startTimestamp = readingTimestamp;
+	} else if (pathname.includes("/manga")) {
+		const isReader = pathname.includes("/chapter"),
+			isAdult = document.querySelector("#premid-adult")?.textContent === "true",
+			mangaTitle = document.querySelector("#premid-manga-title")?.textContent,
+			mangaSlug = document.querySelector("#premid-manga-slug")?.textContent,
+			mangaId = document.querySelector("#premid-manga-id")?.textContent,
+			chapterNumber = document.querySelector(
+				"#premid-chapter-number"
+			)?.textContent,
+			chapterTitle = document.querySelector(
+				"#premid-chapter-title"
+			)?.textContent;
+
+		if (images && !isAdult) {
+			presenceData.largeImageKey = (
+				document.querySelector("#premid-manga-image") as HTMLAnchorElement
+			)?.href;
+		}
+		if (!manga || (isAdult && !adult))
+			presenceData.details = isReader ? "Lendo um Mangá" : "Página de um Mangá";
+		if (manga && ((isAdult && adult) || !isAdult)) {
+			presenceData.details = isReader
+				? `Lendo ${mangaTitle}`
+				: `Página de ${mangaTitle}`;
+		}
+		if (chapter && isReader) {
+			presenceData.state = `Cap. ${chapterNumber}${
+				chapterTitle ? ` - ${chapterTitle}` : ""
+			}`;
+		}
+		if (!chapter && isReader) presenceData.state = "Lendo Capítulo";
+		if (timestamp) presenceData.startTimestamp = readingTimestamp;
+		if (manga && ((isAdult && adult) || !isAdult) && !(chapter && isReader)) {
+			presenceData.buttons = [
+				{
+					label: "Ver Mangá",
+					url: `https://yomumangas.com/manga/${mangaId}/${mangaSlug}`,
```

Comment:
Can't you use `href` for these three buttons?

Reply:
Yeah, Then I'll change the website code to pass a valid href and remove the consts "mangaId" and "mangaSlug"